src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -83,11 +83,14 @@ float64_t RandomFourierDotFeatures::post_dot(float64_t dot_result, index_t par_i
 	return std::cos(dot_result) * constant;
 }
 
-SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
+void RandomFourierDotFeatures::generate_random_params(SGVector<float64_t>& vec)
 {
 	NormalDistribution<float64_t> normal_dist;
 	UniformRealDistribution<float64_t> uniform_real_dist(0.0, 2 * Math::PI);
-	SGVector<float64_t> vec(feats->get_dim_feature_space()+1);
+
+	/* we need space for both coefficients and offset */
```

Comment:
remove, or write something that is appropriate "Ensure passed vector has appropriate size"